### PR TITLE
The breadcrumb should be shown in gisf mode

### PR DESCRIPTION
**Pull request by hatched**
The model switcher was not shown in gisf mode because there was no env and the supported facades has not yet been populated. This branch fixes that by skipping this facade check if the user is in gisf mode and displaying the breadcrumb.

### CODE REVIEW

**Comment by hatched** [comment]
#### To QA

- Deploy the GUI charm with this branch into Juju 2, set the jem-location, gisf, sandbox properly.
- Clear your 'juju' cookies
- Load the GUI and log in
- You should see the breadcrumb with the model switcher with an ugly uuid (#1556)

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/4844/
Test PASSed.

File: jujugui/static/gui/src/app/components/deployment/add-credentials/add-credentials.js [hunk 1]
```diff
@@ -43,7 +43,7 @@ YUI.add('deployment-add-credentials', function() {
             // controller with work to come to generate a controller name.  Also
             // this key will change from 'state-server' to 'controller'.
             // Makyo 2016-03-30
-            'state-server': 'yellow/aws-eu-central',
+            'state-server': 'jcsackett/amazon',
```
Comment (bac) [inline jujugui/static/gui/src/app/components/deployment/add-credentials/add-credentials.js]: whoa, is this a good idea?

File: jujugui/static/gui/src/app/extensions/app-renderer-extension.js [hunk 2]
```diff
@@ -45,13 +45,17 @@ YUI.add('app-renderer-extension', function(Y) {
...
-      if(!env ||
-         env.findFacadeVersion('ModelManager') === null &&
-         env.findFacadeVersion('EnvironmentManager') === null) {
-        // We do not want to show the model switcher if it isn't supported as
-        // it throws an error in the browser console and confuses the user
-        // as it's visible but not functional.
-        showEnvSwitcher = false;
+      // If gisf is enabled then we won't be connected to a model to know
+      // what facades are supported but we can reliably assume it'll be Juju 2
+      // or higher which will support the necessary api calls.
```
Comment (bac) [inline jujugui/static/gui/src/app/extensions/app-renderer-extension.js]: API! API!

File: jujugui/static/gui/src/app/extensions/app-renderer-extension.js [hunk 3]
```diff
@@ -45,13 +45,17 @@ YUI.add('app-renderer-extension', function(Y) {
...
+      if (!this.get('gisf')) {
+        if(!env || env.findFacadeVersion('ModelManager') === null &&
+           env.findFacadeVersion('EnvironmentManager') === null) {
+          // We do not want to show the model switcher if it isn't supported as
+          // it throws an error in the browser console and confuses the user
+          // as it's visible but not functional.
+          showEnvSwitcher = false;
```
Comment (bac) [inline jujugui/static/gui/src/app/extensions/app-renderer-extension.js]: thank you!

File: jujugui/static/gui/src/app/components/user-profile/user-profile.js [hunk 4]
```diff
@@ -95,23 +96,29 @@ YUI.add('user-profile', function() {
...
-          });
-          return;
+        // If gisf is enabled then we won't be connected to a model to know
+        // what facades are supported but we can reliably assume it'll be Juju 2
+        // or higher which will support the necessary api calls.
+        if (! this.props.gisf) {
+          if (!env ||
+            env.findFacadeVersion('ModelManager') === null &&
+            env.findFacadeVersion('EnvironmentManager') === null) {
+            // If we're on Juju < 2 then pass the default model to the list.
```
Comment (bac) [inline jujugui/static/gui/src/app/components/user-profile/user-profile.js]: Shouldn't this check be factored out? You sure do it a bunch.

**Comment by bac** [comment]
Code is 👍 ish, with one suggestion.

File: jujugui/static/gui/src/app/components/deployment/add-credentials/add-credentials.js (hunk 1 above)
Comment (hatched) [inline jujugui/static/gui/src/app/components/deployment/add-credentials/add-credentials.js]: I can remove it from the commit, but neither of them are valid, this is going to be removed in the near future in favour of an api call.

File: jujugui/static/gui/src/app/components/user-profile/user-profile.js (hunk 4 above)
Comment (hatched) [inline jujugui/static/gui/src/app/components/user-profile/user-profile.js]: We could but then we'd have to pass another method deep into these components which would only be a marginal improvement if any.

File: jujugui/static/gui/src/app/components/user-profile/user-profile.js [hunk 5]
```diff
@@ -95,23 +96,29 @@ YUI.add('user-profile', function() {
...
-              // when selected.
-              uuid: '',
-              lastConnection: 'now'
-            }]
-          });
-          return;
+        // If gisf is enabled then we won't be connected to a model to know
+        // what facades are supported but we can reliably assume it'll be Juju 2
+        // or higher which will support the necessary api calls.
+        if (! this.props.gisf) {
```
Comment (huwshimi) [inline jujugui/static/gui/src/app/components/user-profile/user-profile.js]: Whitespace, I'm surprised that even works.

File: jujugui/static/gui/src/app/components/user-profile/user-profile.js [hunk 6]
```diff
@@ -30,6 +30,7 @@ YUI.add('user-profile', function() {
       dbEnvironmentSet: React.PropTypes.func.isRequired,
       env: React.PropTypes.object.isRequired,
       getDiagramURL: React.PropTypes.func.isRequired,
+      gisf: React.PropTypes.bool.isRequired,
```
Comment (huwshimi) [inline jujugui/static/gui/src/app/components/user-profile/user-profile.js]: I see you added an `isRequired`, but I don't see any updates to add this to the tests so I assume there are a bunch of missing prop errors.

File: jujugui/static/gui/src/app/components/user-profile/user-profile.js (hunk 5 above)
Comment (hatched) [inline jujugui/static/gui/src/app/components/user-profile/user-profile.js]: heh woops, JS is not whitespace significant :)

**Comment by bac** [comment]
Followed the QA instructions, mostly. Deployed using GI Joe. Looks good.

File: jujugui/static/gui/src/app/components/deployment/add-credentials/add-credentials.js (hunk 1 above)
Comment (hatched) [inline jujugui/static/gui/src/app/components/deployment/add-credentials/add-credentials.js]: This has been updated.

File: jujugui/static/gui/src/app/components/user-profile/user-profile.js (hunk 6 above)
Comment (hatched) [inline jujugui/static/gui/src/app/components/user-profile/user-profile.js]: Yep you're right, adding those in.

File: jujugui/static/gui/src/app/components/user-profile/user-profile.js (hunk 5 above)
Comment (hatched) [inline jujugui/static/gui/src/app/components/user-profile/user-profile.js]: fixed

File: jujugui/static/gui/src/app/extensions/app-renderer-extension.js (hunk 2 above)
Comment (hatched) [inline jujugui/static/gui/src/app/extensions/app-renderer-extension.js]: fixed

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/4847/
Test FAILed.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/4848/
Test PASSed.

**Comment by hatched** [comment]
:shipit: Thanks for the reviews! QA'd again myself but this branch is now blocking further work so shipping.

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-merge

**Pull request merged**